components/experimental/src/measure/category/mod.rs

```diff
@@ -0,0 +1,39 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::marker::PhantomData;
+
+use crate::measure::measureunit::MeasureUnit;
+
+pub mod area;
+pub mod length;
+pub mod mass;
+pub mod volume;
+
+pub trait MeasureUnitCategory {}
+
+/// A [`MeasureUnit`] that is related to a specific category.
+///
+/// This is useful for type inference and for ensuring that the correct units are used.
+pub struct CategorizedMeasureUnit<T: MeasureUnitCategory> {
```

Comment:
there should be some (unchecked) way of creating a `CategorizedMeasureUnit` from a `MeasureUnit`, otherwise these aren't constructible without compiled data. Something like

```rust
impl MeasureUnit {
  pub fn with_category_unchecked<T: MeasureUnitCategory>(self) -> CategorizedMeasureUnit<T> { ... }
}

Reply:
I think users should use our functions to construct them, and not build their own for now. I do not want someone to build `mass` unit with `meter-per-second`

Reply:
but then it's impossible to build a `CategorizedMeasureUnit` without compiled data; that cannot be the goal

a unit could store its category in data, and `with_category` can fail if it's the wrong one.